### PR TITLE
#1017, #1032, #1115: Neo4j 5.x experimental support

### PR DESCRIPTION
**Pull request by achantavy**
- Removes deprecated `exists()` function with `is not null`. This is compatible with both Neo4j 4.x and 5.x.
- Replaces `CALL db.indexes()` with `SHOW INDEXES` because it was also replaced per https://neo4j.com/docs/operations-manual/5/reference/procedures/#procedure_db_indexes.
- Updates install docs.
- Updates CI to test against both Neo4j 4.4 and 5.latest docker containers.

See
- https://neo4j.com/docs/upgrade-migration-guide/current/version-5/migration/breaking-changes/#_removals.
- #1115.

### CODE REVIEW

**Review by kedarghule — CHANGES_REQUESTED** [review]
I also noticed EXISTS needs to be changed in the below files (hopefully I have a updated version):
1) https://github.com/lyft/cartography/blob/c38aea6c6d6db6d725a63bd031b7af8727eb2afd/cartography/intel/aws/kms.py#L192

2) https://github.com/lyft/cartography/blob/c38aea6c6d6db6d725a63bd031b7af8727eb2afd/cartography/data/jobs/analysis/gcp_gke_basic_auth.json#L5

3) https://github.com/lyft/cartography/blob/c38aea6c6d6db6d725a63bd031b7af8727eb2afd/cartography/data/jobs/analysis/gcp_gke_basic_auth.json#L10

4) https://github.com/lyft/cartography/blob/c38aea6c6d6db6d725a63bd031b7af8727eb2afd/cartography/data/jobs/analysis/aws_ec2_keypair_analysis.json#L6

5) https://github.com/lyft/cartography/blob/c38aea6c6d6db6d725a63bd031b7af8727eb2afd/cartography/data/jobs/analysis/aws_ec2_keypair_analysis.json#L11

File: cartography/data/jobs/analysis/gcp_gke_basic_auth.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     },
     {
       "__comment": "This sets the basic_auth attribute",
```
[the remark this answers is not in the document]
Comment (achantavy) [reply cartography/data/jobs/analysis/gcp_gke_basic_auth.json:9]: Crap, good catch, thank you will fix

**Comment by achantavy** [comment]
@kedarghule - ok! I think I got 'em all. Ran integration tests against a 5.x container and updated the install docs.

**Review by kedarghule — COMMENTED** [review]
@achantavy Looks good! I see one of the integration tests is failing. Should I still go ahead and approve and merge?

**Comment by achantavy** [comment]
@kedarghule I finally was able to add a CI test suite for Neo4j 5! For some reason a manual install of that version kept failing on GitHub actions so I just gave up and used docker.

**Review by kedarghule — DISMISSED** [review]
Sweet :D

**Review by ramonpetgrave64 — APPROVED** [review]

**Comment by achantavy** [comment]
Let's do a follow up PR for a separate docker compose update

**Pull request merged**